tests/onnxruntime/test_modeling.py

```diff
@@ -3099,84 +3114,100 @@ def test_load_vanilla_transformers_which_is_not_supported(self):
 
     @parameterized.expand(grid_parameters({"model_arch": SUPPORTED_ARCHITECTURES, "use_cache": [True]}))
     def test_generate_utils(self, test_name: str, model_arch: str, use_cache: str):
-        if model_arch == "encoder-decoder" and use_cache is True:
-            self.skipTest("encoder-decoder model type with use_cache=True is not supported for bert as a decoder")
         model_args = {"test_name": test_name, "model_arch": model_arch, "use_cache": use_cache}
         self._setup(model_args)
 
-        model_id = MODEL_NAMES[model_arch]
-        model = ORTModelForSeq2SeqLM.from_pretrained(self.onnx_model_dirs[test_name], use_cache=use_cache)
-        tokenizer = get_preprocessor(model_id)
-        text = "This is a sample output"
-        tokens = tokenizer(text, return_tensors="pt")
+        model_ids = self._get_model_ids(model_arch)
+        for model_id in model_ids:
+            if (
+                model_arch == "encoder-decoder"
+                and use_cache is True
```

Comment:
```suggestion
                and use_cache
```

Reply:
I guess this is the common terminology used in the repo, to explicitly check the value?

Reply:
It's me who advocated to use that :D But more and more I agree with Michael, we should not explicitly check `is True` when the type is `bool`, we should only for `Optional[bool]`.

Reply:
I guess this explicit check is something which is also followed in the transformers, I remember Lewis mentioning this, but yeah I agree